### DISCUSSION TITLE
Removing a CLS privilege

### DISCUSSION THREAD

**Discussion by bitnom** [supabase/supabase#20483] [Questions]
> Changing column privileges can break existing queries.
> If you remove a column privilege for a role, that role will lose all access to that column.
> All operations selecting * (including returning * for insert, update, and delete) will fail.

I'm a bit confused by this statement. If I remove which column privilege for a role? I assume that if I remove `SELECT` for a role, that column will no longer be returned to users with that role. However, I don't expect that if I remove `DELETE` for that column/role, the user will no longer be able to `SELECT` it.

Please clear up any ambiguity relating to this message.

**Comment by GaryAustin1** [comment]
I think it is pointing out the there may be side effects to the change.  Any where the column is used in the method you have blocked it will fail and not be silent about it.  So if select is not granted then any operation (select/insert/update/delete) returning that column or the whole row will fail (*).  Points out one of the worst side effects on running code.
